cluster-essentials/pkg/client/install.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package client has APIs for bootstrapping tanzu cluster-essentials on a kubernetes cluster.
+package client
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/client-go/rest"
+)
+
+var downloadBundle = DownloadBundle
+
+const defaultBundleDir = "/tmp"
+const defaultTimeout = 15 * time.Minute
+
+func validatePreInstall(restConfig *rest.Config) error {
+	// TODO implementation
+	return nil
+}
+
+func validatePostInstall(restConfig *rest.Config, timeout time.Duration) error {
+	// TODO implementation
+	return nil
+}
+
+// Install install the cluster essentials packages
+// It take kubernetes rest config of cluster (config), cluster essential bundler path (imgpkgBundlePath)
+// timeout for blocking call (if zero is given, it will use default timeout of 15 min),
+// bundleDestDir to use as temporary dir to download bundles(if empty, it will use /tmp as default path)
+// It return any error encountered
+func Install(config *rest.Config, imgpkgBundlePath string, timeout time.Duration, bundleDestDir string) error {
```

Comment:
Please add `ctx context.Context` as the first parameter and pass it down to functions that require a context.